pkg/mapi2capi/aws.go

```diff
@@ -0,0 +1,431 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mapi2capi
+
+import (
+	"errors"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	mapiv1 "github.com/openshift/api/machine/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/ptr"
+	capav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	errInfrastructureInfrastructureNameCannotBeNil  = errors.New("infrastructure cannot be nil and infrastructure.Status.InfrastructureName cannot be empty")
+	errUnableToFindAMIReference                     = errors.New("unable to find a valid AMI resource reference")
+	errUnableToConvertUnsupportedAMIFilterReference = errors.New("unable to convert AMI Filters reference. Not supported in CAPI")
+	errUnableToConvertUnsupportedAMIARNReference    = errors.New("unable to convert AMI ARN reference. Not supported in CAPI")
```

Comment:
Eventually we will need to do something to resolve these and convert them to IDs to allow this to be converted, but, not yet :)